dbgovl32/dbgovl32.rst

```diff
@@ -0,0 +1,740 @@
+   the executable ELF file.
+
+-  Emit a new ELF section called ``.ARM.debug_overlay`` of type
+   ``SHT_ARM_DEBUG_OVERLAY`` = ``SHT_LOUSER`` + 4 containing a table of
+   entries as follows:
+
+    *debug section offset, debug section index, program section index*
+
+The description earlier in this section shows that the second
+representation can be calculated from the relevant subset of the
+retained relocation data.
+
+GNU ``ld`` has an option (``--emit-relocs``) to retain all relocations in the
+executable file. Clearly this is sufficient.
+
+A better option is to retain only relocations of debug sections (those
+with names matching ``*debug*``) with respect to overlaid program sections
+(``--emit-overlay-debug-relocs``). An overlay-aware linker will readily
+recognize these sections.
+
+For some linkers it might be easier to build a ``.ARM.debug_overlay``
+section directly, as each relocation directive is processed, than to
+emit the original relocations filtered for relevance.
+
+.. raw:: pdf
+
+   PageBreak
+
+The ABI Extension
+=================
+
+We extend the *ABI for the Arm Architecture* (ABI) as noted in this
+section. The extension is optional and no tool chain is required to
+support in order to claim conformance to the ABI. However, tools that
+support debugging overlaid programs should do so in one of the ways
+specified here.
+
+Terminology
+-----------
+
+A linker has two views of a program’s address space that become distinct
+in the presence of *overlaid* program fragments (code or data).
+
+- The *load address* of a program fragment is the address to which a
+  linker expects an external agent such as a program loader, dynamic
+  linker, or debugger to copy the fragment from the ELF file. This is
+  not necessarily the address at which the fragment will execute.
+
+- The *execution address* of a program fragment is the address at which
+  a linker expects the fragment will reside whenever it participates in
+  the program’s execution.
+
+Linker obligations
+------------------
+
+A linker claiming to support the debugging of overlaid programs shall
+ensure the following in the executable ELF files it produces.
+
+-  Each program fragment that overlaps another in the execution address
+   space shall be described by a distinct ELF section header.
+
+-  Target addresses recorded in section header sh\_addr fields and
+   symbol st\_value fields shall be *execution addresses*.
+
+-  Target addresses recorded in p\_paddr fields of program headers of
+   type ``PT_LOAD`` shall be *load addresses.*
+
+-  Each program segment described by a program header PH of type
+   ``PT_LOAD`` shall occupy a different extent [PH.p\_offset, PH.p\_offset
+   + PH.p\_filesz) in the ELF file. (An empty extent shall not overlap
+   any other extent).
+
+In addition, a linker claiming to support debugging of overlaid programs
+shall do *at least one* of the following.
+
+-  | Provide a means to retain all original relocations in the
```

Comment:
this is purely stylistic, but for some reason 'provide..' is placed half a line below the bullet point

Reply:
That seems to be a rst2pdf formatting bug that occurs when you use the prefix `|` to arrange a nonstandard paragraph with known line breaks. Probably the easiest fix is to just stop trying to get that line break in there; it doesn't seem semantic.